utils/s3/client.cc

```diff
@@ -770,7 +782,16 @@ class client::upload_sink final : public client::upload_sink_base {
             co_await upload_part(std::move(_bufs));
         }
         if (upload_started()) {
-            co_await finalize_upload();
+            std::exception_ptr ex;
+            try {
+                co_await finalize_upload();
+            } catch (...) {
+                ex = std::current_exception();
+            }
+            if (ex) {
```

Comment:
It doesn't look correct. What if finalize_upload() throws after failure to `prepare_multipart_upload_parts()`? In that case abort_upload() would be called twice

Reply:
then the same will happen in `future<> multi_part_upload(file&& f, uint64_t total_size, size_t part_size)`. How about removing `abort_upload` from `finalize_upload` altogether and handle it in `catch` in both places, WDYT?

Reply:
Sounds like a plan :)

Reply:
Done